### PR TITLE
add flag to config per_process_gpu_memory_fraction

### PR DESCRIPTION
**Pull request by zhouyoulie**
With "per_process_gpu_memory_fraction" flag (1-100), the users can config per_process_gpu_memory_fraction, otherwise the server process will occupy all the gpu memory.

### CODE REVIEW

File: tensorflow_serving/model_servers/main.cc [hunk 1]
```diff
@@ -346,7 +348,10 @@ int main(int argc, char** argv) {
                        "If non-empty, read an ascii PlatformConfigMap protobuf "
                        "from the supplied file name, and use that platform "
                        "config instead of the Tensorflow platform. (If used, "
-                       "--enable_batching is ignored.)")};
+                       "--enable_batching is ignored.)"),
+      tensorflow::Flag("per_process_gpu_memory_fraction", &per_process_gpu_memory_fraction,
+                       "per_process_gpu_memory_fraction in percent")};
```
Comment (chrisolston) [inline tensorflow_serving/model_servers/main.cc]: Please use a floating point number (so from 0.0 to 1.0).

File: tensorflow_serving/model_servers/main.cc [hunk 2]
```diff
@@ -302,6 +303,7 @@ tensorflow::serving::PlatformConfigMap ParsePlatformConfigMap(
 int main(int argc, char** argv) {
   tensorflow::int32 port = 8500;
   bool enable_batching = false;
+  tensorflow::int32 per_process_gpu_memory_fraction = 100;
```
Comment (chrisolston) [inline tensorflow_serving/model_servers/main.cc]: Please have it default to 0, not 100. (0 is the default value in the tf config proto.)

File: tensorflow_serving/model_servers/main.cc [hunk 3]
```diff
@@ -346,7 +348,10 @@ int main(int argc, char** argv) {
                        "If non-empty, read an ascii PlatformConfigMap protobuf "
                        "from the supplied file name, and use that platform "
                        "config instead of the Tensorflow platform. (If used, "
-                       "--enable_batching is ignored.)")};
+                       "--enable_batching is ignored.)"),
+      tensorflow::Flag("per_process_gpu_memory_fraction", &per_process_gpu_memory_fraction,
```
Comment (chrisolston) [inline tensorflow_serving/model_servers/main.cc]: Please document that the default value is 0, which has special meaning and causes tensorflow to automatically select a value.

File: tensorflow_serving/model_servers/main.cc (hunk 2 above)
Comment (zhouyoulie) [reply tensorflow_serving/model_servers/main.cc]: have fixed~

File: tensorflow_serving/model_servers/main.cc (hunk 3 above)
Comment (zhouyoulie) [reply tensorflow_serving/model_servers/main.cc]: have fixed~

File: tensorflow_serving/model_servers/main.cc (hunk 1 above)
Comment (zhouyoulie) [reply tensorflow_serving/model_servers/main.cc]: have fixed~

**Comment by chrisolston** [comment]
Thanks for the contribution.

**Comment by zhouyoulie** [comment]
nice to learn from u～

**Pull request merged**